### PR TITLE
fix: Enabling SplitControllerAndExtensions leads to incomplete metrics availability

### LINKED ISSUE #3190: Missing metrics data since v1.32.0
**What happened**: Since updating to Agones 1.32.0, Agones isn't producing as many metrics.

**What you expected to happen**: Metrics should be correctly published to the scrape endpoint

**How to reproduce it (as minimally and precisely as possible)**: Not exactly sure but only occurs since updating to the latest version. Also tested on a clean install (minikube on v1.24.9 of k8s) and it appears the same few metrics are being produced from the allocator's prometheus endpoint.

**Anything else we need to know?**: This has only occurred since updating to Agones 1.32.0 from 1.31.0
image

- Agones version: 1.32.0
- Kubernetes version (use `kubectl version`): v1.24.8+k3s1
- Cloud provider or hardware configuration:
- Install method (yaml/helm): helm (upgraded from previous version)

Prometheus endpoint data:
```
# HELP agones_healthcheck_status Current check status (0 indicates success, 1 indicates failure)
# TYPE agones_healthcheck_status gauge
agones_healthcheck_status{check="allocationcache-workerqueue"} 0
agones_healthcheck_status{check="allocator-agones-client"} 0
# HELP agones_k8s_client_http_request_duration_seconds The distribution of HTTP requests latencies to the Kubernetes API by status code
# TYPE agones_k8s_client_http_request_duration_seconds histogram
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="0.01"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="0.025"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="0.05"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="0.075"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="0.1"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="0.25"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="0.5"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="0.75"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="1"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="2"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="3"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/nodes",verb="GET",le="+Inf"} 1
agones_k8s_client_http_request_duration_seconds_sum{endpoint="/api/v1/nodes",verb="GET"} 0.003128939
agones_k8s_client_http_request_duration_seconds_count{endpoint="/api/v1/nodes",verb="GET"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="0.01"} 0
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="0.025"} 0
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="0.05"} 0
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="0.075"} 0
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="0.1"} 0
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="0.25"} 0
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="0.5"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="0.75"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="1"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="2"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="3"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/api/v1/secrets",verb="GET",le="+Inf"} 1
agones_k8s_client_http_request_duration_seconds_sum{endpoint="/api/v1/secrets",verb="GET"} 0.292480337
agones_k8s_client_http_request_duration_seconds_count{endpoint="/api/v1/secrets",verb="GET"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="0.01"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="0.025"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="0.05"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="0.075"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="0.1"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="0.25"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="0.5"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="0.75"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="1"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="2"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="3"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/agones.dev/v1/gameservers",verb="GET",le="+Inf"} 1
agones_k8s_client_http_request_duration_seconds_sum{endpoint="/apis/agones.dev/v1/gameservers",verb="GET"} 0.004204918
agones_k8s_client_http_request_duration_seconds_count{endpoint="/apis/agones.dev/v1/gameservers",verb="GET"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="0.01"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="0.025"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="0.05"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="0.075"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="0.1"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="0.25"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="0.5"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="0.75"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="1"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="2"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="3"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET",le="+Inf"} 1
agones_k8s_client_http_request_duration_seconds_sum{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET"} 0.003079518
agones_k8s_client_http_request_duration_seconds_count{endpoint="/apis/multicluster.agones.dev/v1/gameserverallocationpolicies",verb="GET"} 1
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="0.01"} 89969
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="0.025"} 90032
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="0.05"} 90058
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="0.075"} 90059
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="0.1"} 90059
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="0.25"} 90059
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="0.5"} 90059
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="0.75"} 90059
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="1"} 90059
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="2"} 90059
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="3"} 90059
agones_k8s_client_http_request_duration_seconds_bucket{endpoint="/version",verb="GET",le="+Inf"} 90059
agones_k8s_client_http_request_duration_seconds_sum{endpoint="/version",verb="GET"} 157.5060817110015
agones_k8s_client_http_request_duration_seconds_count{endpoint="/version",verb="GET"} 90059
# HELP agones_k8s_client_http_request_total The total of HTTP requests to the Kubernetes API by status code
# TYPE agones_k8s_client_http_request_total counter
agones_k8s_client_http_request_total{status_code="200",verb="GET"} 92261
# HELP agones_k8s_client_workqueue_longest_running_processor How long the longest running workqueue processor has been running in microseconds.
# TYPE agones_k8s_client_workqueue_longest_running_processor gauge
agones_k8s_client_workqueue_longest_running_processor{queue_name=""} 0
# HELP agones_k8s_client_workqueue_unfinished_work_seconds How long unfinished work has been sitting in the workqueue in seconds.
# TYPE agones_k8s_client_workqueue_unfinished_work_seconds gauge
agones_k8s_client_workqueue_unfinished_work_seconds{queue_name=""} 0
# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 3.345e-05
go_gc_duration_seconds{quantile="0.25"} 4.333e-05
go_gc_duration_seconds{quantile="0.5"} 4.515e-05
go_gc_duration_seconds{quantile="0.75"} 4.9271e-05
go_gc_duration_seconds{quantile="1"} 0.004041623
go_gc_duration_seconds_sum 0.146905161
go_gc_duration_seconds_count 2075
# HELP go_goroutines Number of goroutines that currently exist.
# TYPE go_goroutines gauge
go_goroutines 159
# HELP go_info Information about the Go environment.
# TYPE go_info gauge
go_info{version="go1.19.1"} 1
# HELP go_memstats_alloc_bytes Number of bytes allocated and still in use.
# TYPE go_memstats_alloc_bytes gauge
go_memstats_alloc_bytes 2.634016e+07
# HELP go_memstats_alloc_bytes_total Total number of bytes allocated, even if freed.
# TYPE go_memstats_alloc_bytes_total counter
go_memstats_alloc_bytes_total 8.344807544e+09
# HELP go_memstats_buck_hash_sys_bytes Number of bytes used by the profiling bucket hash table.
# TYPE go_memstats_buck_hash_sys_bytes gauge
go_memstats_buck_hash_sys_bytes 6163
# HELP go_memstats_frees_total Total number of frees.
# TYPE go_memstats_frees_total counter
go_memstats_frees_total 4.8946048e+07
# HELP go_memstats_gc_sys_bytes Number of bytes used for garbage collection system metadata.
# TYPE go_memstats_gc_sys_bytes gauge
go_memstats_gc_sys_bytes 1.2388848e+07
# HELP go_memstats_heap_alloc_bytes Number of heap bytes allocated and still in use.
# TYPE go_memstats_heap_alloc_bytes gauge
go_memstats_heap_alloc_bytes 2.634016e+07
# HELP go_memstats_heap_idle_bytes Number of heap bytes waiting to be used.
# TYPE go_memstats_heap_idle_bytes gauge
go_memstats_heap_idle_bytes 3.8739968e+07
# HELP go_memstats_heap_inuse_bytes Number of heap bytes that are in use.
# TYPE go_memstats_heap_inuse_bytes gauge
go_memstats_heap_inuse_bytes 3.0138368e+07
# HELP go_memstats_heap_objects Number of allocated objects.
# TYPE go_memstats_heap_objects gauge
go_memstats_heap_objects 36839
# HELP go_memstats_heap_released_bytes Number of heap bytes released to OS.
# TYPE go_memstats_heap_released_bytes gauge
go_memstats_heap_released_bytes 3.3939456e+07
# HELP go_memstats_heap_sys_bytes Number of heap bytes obtained from system.
# TYPE go_memstats_heap_sys_bytes gauge
go_memstats_heap_sys_bytes 6.8878336e+07
# HELP go_memstats_last_gc_time_seconds Number of seconds since 1970 of last garbage collection.
# TYPE go_memstats_last_gc_time_seconds gauge
go_memstats_last_gc_time_seconds 1.6854462085508342e+09
# HELP go_memstats_lookups_total Total number of pointer lookups.
# TYPE go_memstats_lookups_total counter
go_memstats_lookups_total 0
# HELP go_memstats_mallocs_total Total number of mallocs.
# TYPE go_memstats_mallocs_total counter
go_memstats_mallocs_total 4.8982887e+07
# HELP go_memstats_mcache_inuse_bytes Number of bytes in use by mcache structures.
# TYPE go_memstats_mcache_inuse_bytes gauge
go_memstats_mcache_inuse_bytes 14400
# HELP go_memstats_mcache_sys_bytes Number of bytes used for mcache structures obtained from system.
# TYPE go_memstats_mcache_sys_bytes gauge
go_memstats_mcache_sys_bytes 15600
# HELP go_memstats_mspan_inuse_bytes Number of bytes in use by mspan structures.
# TYPE go_memstats_mspan_inuse_bytes gauge
go_memstats_mspan_inuse_bytes 253640
# HELP go_memstats_mspan_sys_bytes Number of bytes used for mspan structures obtained from system.
# TYPE go_memstats_mspan_sys_bytes gauge
go_memstats_mspan_sys_bytes 522240
# HELP go_memstats_next_gc_bytes Number of heap bytes when next garbage collection will take place.
# TYPE go_memstats_next_gc_bytes gauge
go_memstats_next_gc_bytes 5.3143576e+07
# HELP go_memstats_other_sys_bytes Number of bytes used for other system allocations.
# TYPE go_memstats_other_sys_bytes gauge
go_memstats_other_sys_bytes 2.781485e+06
# HELP go_memstats_stack_inuse_bytes Number of bytes in use by the stack allocator.
# TYPE go_memstats_stack_inuse_bytes gauge
go_memstats_stack_inuse_bytes 2.424832e+06
# HELP go_memstats_stack_sys_bytes Number of bytes obtained from system for stack allocator.
# TYPE go_memstats_stack_sys_bytes gauge
go_memstats_stack_sys_bytes 2.424832e+06
# HELP go_memstats_sys_bytes Number of bytes obtained from system.
# TYPE go_memstats_sys_bytes gauge
go_memstats_sys_bytes 8.7017504e+07
# HELP go_threads Number of OS threads created.
# TYPE go_threads gauge
go_threads 19
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 223.9
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 1.048576e+06
# HELP process_open_fds Number of open file descriptors.
# TYPE process_open_fds gauge
process_open_fds 20
# HELP process_resident_memory_bytes Resident memory size in bytes.
# TYPE process_resident_memory_bytes gauge
process_resident_memory_bytes 7.3662464e+07
# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1.68520060843e+09
# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 8.4731904e+08
# HELP process_virtual_memory_max_bytes Maximum amount of virtual memory available in bytes.
# TYPE process_virtual_memory_max_bytes gauge
process_virtual_memory_max_bytes 1.8446744073709552e+19```

### LINKED ISSUE #3244: Bug in the ServiceMap config of the SplitControllerAndExtensions feature
**What happened**:
While deploying The Agones Helmchart (v1.32.0) and dashboards (from main) I was unable to display any data.

**What you expected to happen**:
Metrics would come in to Prometheus and data would be shown on the Grafana dashboard.

**How to reproduce it (as minimally and precisely as possible)**:
Deploy Agones and prometheus-stack charts, and deploy the the dashboard configmaps.

**Anything else we need to know?**:
Have resolved the issue by adding the following featureGate setting.
`agones.featureGates: "SplitControllerAndExtensions=false"`

**Environment**:
- Agones version: 1.32.0
- Kubernetes version (use `kubectl version`): v1.27.1
- Cloud provider or hardware configuration: AKS
- Install method (yaml/helm): helm
- Troubleshooting guide log(s):
I have traced the issue back to SplitControllerAndExtensions featureGate, this feature updates the agones-controller-service label to use 'agones.dev/role: extensions' instead of 'agones.dev/role: controller'. While the ServiceMonitor 'agones-controller-monitor' still remains configured with the matchLabels 'agones.dev/role: controller'. This prevents Prometheus of finding any service to scrape, so no data is gathered. After adding the override `agones.featureGates: "SplitControllerAndExtensions=false"` to the helm values, the label of the service reverted to controller and Prometheus was able to discover the service and scrape metrics.
- Others:

### PR DESCRIPTION
**Pull request by aimuz**
Signed-off-by: aimuz <mr.imuz@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

/kind bug

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3244 
Closes #3190

**Special notes for your reviewer**:

The PR fix is to create a new service dedicated to providing metrics, as the extensions metrics are very limited and a lot of metrics are still in the controller.

### CODE REVIEW

File: install/yaml/install.yaml [hunk 1]
```diff
@@ -16435,39 +16501,39 @@ spec:
           readOnly: true
 ---
 # Source: agones/templates/extensions.yaml
-apiVersion: apiregistration.k8s.io/v1
```
Comment (aimuz) [inline install/yaml/install.yaml:16438]: Strange diff changes, they are exactly the same

File: install/yaml/install.yaml [hunk 2]
```diff
@@ -16435,39 +16501,39 @@ spec:
...
   version: v1
 ---
 # Source: agones/templates/extensions.yaml
-# Copyright 2018 Google LLC All Rights Reserved.
```
Comment (aimuz) [inline install/yaml/install.yaml:16459]: Strange diff changes, they are exactly the same

**Comment by aimuz** [comment]
@markmandel Can you take a look at this?

**Comment by aimuz** [comment]
ref: https://github.com/googleforgames/agones/issues/3190

**Comment by markmandel** [comment]
Calling @chiayi - you were looking into this stuff, yeah?

**Comment by markmandel** [comment]
In which case, I defer review to you @chiayi 😄

**Comment by aimuz** [comment]
In our environment, this works fine and we can see the relevant metrics

**Comment by chiayi** [comment]
Sorry for the delay, I was able to run it as well and it works fine LGTM

**Review by chiayi — APPROVED** [review]

**Comment by markmandel** [comment]
You'll need to run `make gen-install` please and update. Since the install.yaml changed because of the Helm changes.

**Comment by aimuz** [comment]
> You'll need to run make gen-install please and update. Since the install.yaml changed because of the Helm changes.

thank you, I'm done

**Review by chiayi — APPROVED** [review]

**Comment by markmandel** [comment]
@chiayi just triple checking before I approve and merge - this looks good to go on your end?

**Comment by markmandel** [comment]
Ignore me - you already approved 👍🏻

**Review by markmandel — APPROVED** [review]

**Comment by aimuz** [comment]
Is there anything else this PR needs to do to merge?

**Pull request merged**